File: DP_CS_Code_DBhalwani/tools.js
function isEven(a) {

	if (a % 2 == 0) {
		return true;
	}
		return false;
}
	

console.log("")
console.log("isEven")
console.log('Is 0 even?')
console.log(isEven(0))
console.log('Is 15 even?')
console.log(isEven(15))



function missing_char(str, n) {
	var new1 = str.slice(0, n);
	var new2 = str.slice(n+1);
	var new_str = new1 + new2;
	return new_str;
}


console.log("")
console.log("missing_char")
console.log(missing_char('Converge', 3))
console.log(missing_char('metal', 2))



function base2To10(str) {
	var total = 0;
	var i = 0;
	var length = str.length;
	while (i < length) {
		total = total + Number(str.charAt(length - i - 1)) * 2 ** i;

		i = i + 1
		
	}
		
	return total

}
	


console.log("")
console.log('base2To10')
console.log(base2To10('1011'))
console.log(base2To10('111111'))




function sumDigits(a) {
	var total = 0;
	var i = 0;
	var a = String(a);
	while (i < a.length) {
		total = total + Number(a.charAt(i));
		i = i + 1;
	}

	return total

}



console.log("")
console.log('sumDigits')
console.log(sumDigits('45'))
console.log(sumDigits('101'))
console.log(sumDigits('3'))



function scaleElementsA(a, b) {
	var i = 0;
	while (i < b.length) {
		b[i] = b[i] * a;

		i = i + 1;
	}

	return b
}


console.log("")
console.log('scaleElementsA')
console.log(scaleElementsA(2, [1,2,3]))
console.log(scaleElementsA(3, [4,3]))
console.log(scaleElementsA(0, [4,5,6,7]))




function scaleElementsB(a, b) {
	var i = 0;
	var z = [];
	while (i < b.length) {
		z = z.concat(b[i] * a);

		i = i + 1;
	}

	return z
}


console.log("")
console.log('scaleElementsB')
console.log(scaleElementsB(2, [1,2,3]))
console.log(scaleElementsB(3, [4,3]))
console.log(scaleElementsB(0, [4,5,6,7]))



function addStringsSmallLarge(a,b) {

	if (a.length < b.length) {
		return b + a;
	}

	else {
		return a + b;
	}
}

		
console.log('')
console.log('addStringsSmallLarge')
console.log(addStringsSmallLarge('aaa', 'bb'))
console.log(addStringsSmallLarge('aaa', 'bbbb'))
console.log(addStringsSmallLarge('aaa', 'bbb'))

File: DP_CS_Code_DBhalwani/tools.py
def isEven(a):
	if a % 2 == 0:
		return True
	else:
		return False


print("")
print('isEven')
print('Is 0 even?')
print(isEven(0))
print('Is 15 even?')
print(isEven(15))


print("")



def missing_char(str, n):
  return str[0:n] + str[n+1:len(str)]

print('missingChar')
print(missing_char('Converge', 3))
print(missing_char('metal', 2))


print("")



def base2To10(str):
	total = 0
	i = 1
	while i < len(str):
		total = total + int(str[len(str)-i])*2^(i-1)
		i = i + 1
	return total

print('base2To10')
print(base2To10('101'))
print(base2To10('111111'))



print("")




def sumDigits(a):
	total = 0
	i = 0
	a = str(a) #casting is the process of changing type
	#alternatively: for i in range (0, len(a), 1):
	#count (0, the number we start at), check (len(a), the maximum value, change (1, the value we increment by))

	while i < len(str(a)):
		total = total + int(a[i])
		i = i + 1
	
	return total

print('sumDigits')
print(sumDigits(45))
print(sumDigits(101))
print(sumDigits(3))


print("")



def sumDigitsVersion2(a):
	total = 0

	while (a > 0):
		total = total + a % 10
		a = a // 10 #integer division, which divides and chops off the decimals
		# 367 / 10 = 36.7, but 367 // 10 = 36
	
	return total

print('sumDigitsVersion2')
print(sumDigitsVersion2(45))
print(sumDigitsVersion2(101))
print(sumDigitsVersion2(3))



print("")



def scaleElementsA(a,b):
	i = 0
	for i in range (0, len(b), 1):
		b[i] = b[i] * a
	return b


print('scaleElementsA')
print(scaleElementsA(2, [1,2,3]))
print(scaleElementsA(3, [4,3]))
print(scaleElementsA(0, [4,5,6,7]))



print("")



def scaleElementsB(a,b):
	i = 0
	z = []
	for i in range (0, len(b), 1):
		z.append(b[i] * a)
		z = z
	return z


print('scaleElementsB')
print(scaleElementsB(2, [1,2,3]))
print(scaleElementsB(3, [4,3]))
print(scaleElementsB(0, [4,5,6,7]))



print("")





def addStringsSmallLarge(a,b):
	if len(a) < len(b):
		return b + a
	else:
		return a + b


print('addStringsSmallLarge')
print(addStringsSmallLarge('aaa', 'bb'))
print(addStringsSmallLarge('aaa', 'bbbb'))
print(addStringsSmallLarge('aaa', 'bbb'))



print("")